src/loaders/compute_tools/microtrait/microtrait_util.py

```diff
@@ -48,21 +107,14 @@ def create_trait_unwrapped_rules(
         for rule_name in rule_names.copy():
             unwrapped_rule = rule_dict.get(rule_name)
             if unwrapped_rule:
-                # find and extract gene names from the unwrapped rule expression.
-                # i.e. "('pcbA' | 'pcbB' | 'pcbC' | 'pcbD' | 'pcbE' | 'pcbF' | 'pcbG' | 'pcbH')"
-                gene_names = set(re.findall(r"'(\w+)'", unwrapped_rule))
+                gene_names = _retrieve_genes(unwrapped_rule)
                 # remove the rule name from the set and add the unwrapped rule expression
                 trait_rule_unwrapped_mapping[trait_name].remove(rule_name)
                 trait_rule_unwrapped_mapping[trait_name].update(gene_names)
             else:
                 raise ValueError(f"Rule name {rule_name} not found in ruleunwrapped file")
 
     if trait_unwrapped_rules_file is not None:
-        with open(trait_unwrapped_rules_file, 'w') as file:
-            file.write(f"{SYS_TRAIT_ID}\t{UNWRAPPED_GENE_COL}\n")
-            for trait, rule_set in trait_rule_unwrapped_mapping.items():
-                # Join the set elements into a single string
-                unwrapped_rules = ';'.join(rule_set)
-                file.write(f"{trait}\t{unwrapped_rules}\n")
+        _write_trait_unwrapped_rules(trait_unwrapped_rules_file, trait_rule_unwrapped_mapping)
 
     return trait_rule_unwrapped_mapping
```

Comment:
Is there some way we could verify we're doing this correctly other than via code review? Like a dataset we could compare our results to?

Reply:
I hope we do. I am not aware of such things, perhaps ask Ulas? But if we do, we can just use that instead of creating our own logic.

Reply:
We should ask @psdehal for a sanity check when the data is loaded. With any luck Ulas will add equivalent functionality to microtrait and like you say, we can just use that